File: packages/python-sdk/strait/operations/api_keys.py
"""API key management operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class APIKeysService(BaseService):
    def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/api-keys", query=query)

    def create(self, body: Any) -> dict[str, Any]:
        return self._request("POST", "/v1/api-keys", body=body)

    def delete(self, key_id: str) -> dict[str, Any]:
        return self._request(
            "DELETE", "/v1/api-keys/{keyID}", path_params={"keyID": key_id},
        )

    def rotate(self, key_id: str, body: Any) -> dict[str, Any]:
        return self._request(
            "POST", "/v1/api-keys/{keyID}/rotate", path_params={"keyID": key_id}, body=body,
        )


class AsyncAPIKeysService(AsyncBaseService):
    async def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/api-keys", query=query)

    async def create(self, body: Any) -> dict[str, Any]:
        return await self._request("POST", "/v1/api-keys", body=body)

    async def delete(self, key_id: str) -> dict[str, Any]:
        return await self._request(
            "DELETE", "/v1/api-keys/{keyID}", path_params={"keyID": key_id},
        )

    async def rotate(self, key_id: str, body: Any) -> dict[str, Any]:
        return await self._request(
            "POST", "/v1/api-keys/{keyID}/rotate", path_params={"keyID": key_id}, body=body,
        )

File: packages/ruby-sdk/lib/strait/operations/api_keys.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for API key management.
    class APIKeysService < BaseService
      # List all API keys.
      # GET /v1/api-keys
      def list(query: nil)
        _request(:get, "/v1/api-keys", query: query)
      end

      # Create a new API key.
      # POST /v1/api-keys
      def create(body)
        _request(:post, "/v1/api-keys", body: body)
      end

      # Delete an API key.
      # DELETE /v1/api-keys/{keyID}
      def delete(key_id)
        _request(:delete, "/v1/api-keys/{keyID}", path_params: { "keyID" => key_id })
        nil
      end

      # Rotate an API key.
      # POST /v1/api-keys/{keyID}/rotate
      def rotate(key_id, body)
        _request(:post, "/v1/api-keys/{keyID}/rotate", path_params: { "keyID" => key_id }, body: body)
      end
    end
  end
end
